CMakeLists.txt

```diff
@@ -1,78 +1,654 @@
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/Toy/data/languages/toy_builder_le.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/Toy/data/languages/toy_le.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/Toy/data/languages/toy_wsz_be.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/Toy/data/languages/toy_wsz_le.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/tricore/data/languages/tricore.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/V850/data/languages/V850.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/x86/data/languages/x86-64.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/x86/data/languages/x86.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/Z80/data/languages/z180.slaspec"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors/Z80/data/languages/z80.slaspec"
+)
+
+set(sla_spec_output "${CMAKE_CURRENT_BINARY_DIR}/Processors")
+
+add_custom_command(
+  OUTPUT "${sla_spec_output}"
+  DEPENDS ${spec_file_list}
+  COMMAND "${CMAKE_COMMAND}" -E rm -rf "${sla_spec_output}"
+  COMMAND "${CMAKE_COMMAND}" -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/src/Ghidra/Processors" "${sla_spec_output}"
+  COMMENT "sleigh: Copying the spec files"
+  VERBATIM
+)
+
+add_custom_target(sla_spec_output_builder ALL DEPENDS
+  "${sla_spec_output}"
+)
+
+set(sla_spec_build_log "${sla_spec_output}/build.log")
+
+add_custom_command(
+  OUTPUT "${sla_spec_build_log}"
+  DEPENDS ${spec_file_list}
+  COMMAND sleigh_opt -a "${sla_spec_output}" > "${sla_spec_build_log}" 2>&1
+  WORKING_DIRECTORY "${sla_spec_output}"
+  COMMENT "sleigh: Compiling the spec files (${sla_spec_build_log})"
+  VERBATIM
+)
+
+add_custom_target(sleigh_specs ALL DEPENDS
+  "${sla_spec_build_log}"
+)
+
+add_dependencies(sleigh_specs
+  sla_spec_output_builder
+)
+
+#
+# Examples
+#
+
+if(SLEIGH_ENABLE_EXAMPLES)
+  add_executable(sleighexample
+    "${library_root}/sleighexample.cc"
   )
 
-configure_file(
-  "${PROJECT_SOURCE_DIR}/cmake/sleighConfig.cmake.in"
-  "${sleigh_cmake_config}"
-  @ONLY
+  target_link_libraries(sleighexample PRIVATE
+    sla
+    decomp
   )
 
-install(
-  FILES "${sleigh_cmake_config}"
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/sleigh"
+  add_custom_target(sleighexample_runner
+    "${CMAKE_COMMAND}" -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/specfiles"
+    COMMAND "${CMAKE_COMMAND}" -E create_symlink "${CMAKE_CURRENT_BINARY_DIR}/Processors/x86/data/languages/x86.sla" "${CMAKE_CURRENT_BINARY_DIR}/specfiles/x86.sla"
+    COMMAND sleighexample disassemble
+    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
   )
 
-install(
-  EXPORT sleighTargets
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/sleigh"
+  add_dependencies(
+    sleighexample_runner
+    sleigh_specs
   )
+endif()
+
+#
+# Install targets
+#
+
+if(SLEIGH_ENABLE_INSTALL)
+  # TODO: Install each target
```

Comment:
I can give these `TODO` comments a try if you don't have time.

Reply:
Ok, I see you've done the `TODO` comments now. Still, if there's a way I can help, let me know.